### PR TITLE
fix(storage-browser): invalid list ui state on error

### PR DESCRIPTION
**Pull request by pranavosu**
#### Description of changes

Use `hasError` to determine table visibility. This prevents being able to continually click through the same folder where an error occurred. 

nav-bug

In the future, we would want to reconsider combining error states with valid data in [useDataState](https://github.com/aws-amplify/amplify-ui/blob/9ac7b8d776bbb605a2e9283bb55ad5fdbbb59c09/packages/react-core/src/hooks/useDataState.ts#L43). 

#### Issue #, if available

https://app.asana.com/0/1208039352063378/1208383276287447

#### Description of how you validated changes

Create a folder with Deny perms in S3 bucket. Navigate to it in storage browser. Verify that you cannot keep clicking on the same folder.

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by AllanZhengYP — DISMISSED** [review]

File: packages/react-storage/src/components/StorageBrowser/views/LocationsView/__tests__/LocationsView.spec.tsx [hunk 1]
```diff
@@ -125,6 +131,16 @@ describe('LocationsListView', () => {
     const messageText = screen.getByText(errorMessage);
     expect(message).toBeInTheDocument();
     expect(messageText).toBeInTheDocument();
+
+    // table doesn't render
+    const table = screen.queryByRole('table');
+    expect(table).not.toBeInTheDocument();
+
+    // pagination disabled
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/views/LocationsView/__tests__/LocationsView.spec.tsx:139]: Appreciate the comments ⭐

**Review by calebpollman — APPROVED** [review]
Thanks for handling this! 

Approach makes sense but have some thoughts that can be addressed as follow ups - afaict in the error scenario addressed here the state of `LocationsView` and `LocationDetailView` becomes more or less unusable with the exception of the refresh button and breadcrumbs navigation (detail view only). Wondering if we should provide a descriptive error view inside the `Table` (rather than hiding it entirely) with a retry button to improve the UX? Thinking we need product input from @dbanksdesign here

**Comment by pranavosu** [comment]
> Thanks for handling this!
> 
> Approach makes sense but have some thoughts that can be addressed as follow ups - afaict in the error scenario addressed here the state of `LocationsView` and `LocationDetailView` becomes more or less unusable with the exception of the refresh button and breadcrumbs navigation (detail view only). Wondering if we should provide a descriptive error view inside the `Table` (rather than hiding it entirely) with a retry button to improve the UX? Thinking we need product input from @dbanksdesign here

I agree! We aren't making the best of use of screen real estate. I'm not sure it has to be a part of the Table component, but we could use a more useful Error / Empty State view.

**Review by calebpollman — APPROVED** [review]

**Review by AllanZhengYP — APPROVED** [review]

**Comment by reesscot** [comment]
@pranavosu Can you please push this to a branch in the amplify-ui repo rather than using a fork? Our workflows aren't working right for forks right now

**Comment by pranavosu** [comment]
> @pranavosu Can you please push this to a branch in the amplify-ui repo rather than using a fork? Our workflows aren't working right for forks right now

I created a [new PR](https://github.com/aws-amplify/amplify-ui/pull/5889) and the workflows in this one started working. I think it's because the commit hash is identical i.e. 
9ef4ebb. I'm just going to merge this PR if the tests pass, and move the rest of my work to a branch from my fork for next time.

**Pull request merged**